go/api/config.go

```diff
@@ -0,0 +1,290 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+import "github.com/aws/glide-for-redis/go/glide/protobuf"
+
+const (
+	defaultHost = "localhost"
+	defaultPort = 6379
+)
+
+// NodeAddress represents the host address and port of a node in the cluster.
+type NodeAddress struct {
+	Host string // Optional: if not supplied, "localhost" will be used.
+	Port int    // Optional: if not supplied, 6379 will be used.
+}
+
+func (addr *NodeAddress) toProtobuf() *protobuf.NodeAddress {
+	if addr.Host == "" {
+		addr.Host = defaultHost
+	}
+
+	if addr.Port == 0 {
+		addr.Port = defaultPort
+	}
+
+	return &protobuf.NodeAddress{Host: addr.Host, Port: uint32(addr.Port)}
+}
+
+// RedisCredentials represents the credentials for connecting to a Redis server.
+type RedisCredentials struct {
+	// Optional: the username that will be used for authenticating connections to the Redis servers. If not supplied, "default"
+	// will be used.
+	username string
+	// Required: the password that will be used for authenticating connections to the Redis servers.
+	password string
+}
+
+// NewRedisCredentials returns a new RedisCredentials struct with the given username and password.
+func NewRedisCredentials(username string, password string) *RedisCredentials {
```

Comment:
since the zero-values for the struct are all suitable for defaults, if we wanted we could get rid of the whole builder and just have the user use the struct directly. But that solution may be slightly more awkward in some ways, eg forming the address list